src/fmeval/eval_algorithms/answer_relevance.py

```diff
@@ -0,0 +1,225 @@
+import logging
+from typing import Optional, List, Dict, Any, Union
+
+import numpy as np
+
+from fmeval.constants import (
+    DatasetColumns,
+    MEAN,
+)
+from fmeval.data_loaders.util import get_dataset
+from fmeval.data_loaders.data_config import DataConfig
+from fmeval.eval_algorithms.save_strategy import SaveStrategy
+from fmeval.eval_algorithms.util import get_dataset_configs
+from fmeval.eval_algorithms.common import evaluate_dataset
+from fmeval.eval_algorithms.eval_algorithm import EvalAlgorithmInterface, EvalAlgorithmConfig
+from fmeval.eval_algorithms import (
+    EvalAlgorithm,
+    EvalOutput,
+    EvalScore,
+)
+from fmeval.eval_algorithms.util import validate_dataset
+from fmeval.model_runners.model_runner import ModelRunner
+from fmeval.transforms.common import GetModelOutputs, GeneratePrompt
+from fmeval.transforms.transform import Transform
+from fmeval.transforms.transform_pipeline import TransformPipeline
+from fmeval.transforms.util import validate_call
+from fmeval.util import get_eval_results_path
+
+logger = logging.getLogger(__name__)
+
+ANSWER_RELEVANCE = EvalAlgorithm.ANSWER_RELEVANCE.value
+
+QUESTION_GENERATION_PROMPT = "question_generation_prompt"
+ANSWER = "answer"
+RAW_GEN_QUESTION = "raw_gen_question"
+
+QUESTION_GEN_PROMPT = """\
+Human: Generate question for the given answer.
+Answer:\nThe PSLV-C56 mission is scheduled to be launched on Sunday, 30 July 2023 at 06:30 IST / 01:00 UTC. It will be launched from the Satish Dhawan Space Centre, Sriharikota, Andhra Pradesh, India
+Question: When is the scheduled launch date and time for the PSLV-C56 mission, and where will it be launched from?
+
+Answer:$answer
+Question: Assistant:"""  # noqa: E501
+
+
+class AnswerRelevanceScore(Transform):
+    """This transform augments its input record with the computed answer relevance score.
+
+    See the docstring for `Answer Relevance` for more details regarding the score itself.
+    """
+
+    def __init__(
+        self,
+        embeddings_model: ModelRunner,
+        model_output_key: str = DatasetColumns.MODEL_OUTPUT.value.name,
+        output_key: str = ANSWER_RELEVANCE,
+    ):
+        """AnswerRelevanceScore initializer.
+
+        :param embeddings_model: An instance of ModelRunner representing the embedding model to be used.
+        :param model_output_key: The key corresponding to the model output.
+        :param output_key: The key corresponding to the answer relevance score that will be added to the input record.
+        """
+        super().__init__(output_key=output_key, embeddings_model=embeddings_model)
```

Comment:
Make sure to include `model_output_key` as well. For the sake of clarity, it may be better to have`embeddings_model` come first as a positional argument instead of reordering it and making it a kwarg.